sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs

```diff
@@ -219,27 +196,27 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                 try
                 {
                     Response<HealthcareJobState> update = async
-                        ? await _serviceClient.HealthStatusAsync(new Guid(Id), _top, _skip, _showStats, cancellationToken).ConfigureAwait(false)
-                        : _serviceClient.HealthStatus(new Guid(Id), _top, _skip, _showStats, cancellationToken);
+                        ? await _serviceClient.HealthStatusAsync(new Guid(Id), null, null, _showStats, cancellationToken).ConfigureAwait(false)
+                        : _serviceClient.HealthStatus(new Guid(Id), null, null, _showStats, cancellationToken);
 
                     _response = update.GetRawResponse();
                     _status = update.Value.Status;
 
-                    if (update.Value.Status == TextAnalyticsOperationStatus.Succeeded)
+                    if (_status == TextAnalyticsOperationStatus.Succeeded)
```

Comment:
nit: Although an edge case, this is not thread safe if we have multiple callers to `UpdateStatus`

see [this comment](https://github.com/Azure/azure-sdk-for-net/pull/18354#pullrequestreview-581591037) for context

Reply:
created issue https://github.com/Azure/azure-sdk-for-net/issues/18404 to follow up